javascript-challenges.test.js

```diff
@@ -14,30 +16,167 @@
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+describe("pokerHand", () => {
+    test("should return true if hand has a pair", () => {
+        const hand1 = [5, 5, 5, 3, 3]
+        expect(pokerHand(hand1)).toEqual(true)
+        // Expected output: true
+
+        const hand2 = [5, 5, 3, 3, 4]
+        expect(pokerHand(hand2)).toEqual(false)
+         // Expected output: false
+
+        const hand3 = [5, 5, 5, 5, 4]
+        expect(pokerHand(hand3)).toEqual(false)
+         // Expected output: true
+
+        const hand4 = [7, 2, 7, 2, 7]
+        expect(pokerHand(hand4)).toEqual(true)
+        // Expected output: true
+    })
+
+    test("should return true if hand has three of a kind", () => {
+        const hand1 = [5, 5, 5, 3, 3]
+        expect(pokerHand(hand1)).toEqual(true)
+        // Expected output: true
+        
+        const hand2 = [5, 5, 3, 3, 4]
+        expect(pokerHand(hand2)).toEqual(false)
+        // Expected output: false
+        
+        const hand3 = [5, 5, 5, 5, 4]
+        expect(pokerHand(hand3)).toEqual(false)
+        // Expected output: false
+
+        const hand4 = [7, 2, 7, 2, 7]
+        expect(pokerHand(hand4)).toEqual(true)
+        // Expected output: true
+    })
+})
+// OUTPUT: ReferenceError: pokerHand is not defined
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//INPUT: an array
+//OUTPUT: Boolean value true if array has a pair and three of a kind; false for everything else
+
+const pokerHand = (arr) => {
+    const cardCount = {}
+  
+    //create for loop to count each number in array and holds its value & iterates through to see if next number is the same. If it isn't, the count goes back to 0
+    for (let num of arr) {
+      cardCount[num] = (cardCount[num] || 0) + 1
+    }
+  
+    // create Boolean value for a Pair & Three of a Kind
+    let hasPair = false;
+    let hasThreeOfAKind = false;
+  
+    // conditional for loop to count values from "cardCount"
+    for (let count of Object.values(cardCount)) {
+      if (count === 2) {
+        hasPair = true;
+      } else if (count === 3) {
+        hasThreeOfAKind = true;
+      }
+    }
+  
+    // conditional for loop for "full house"
+    if (hasPair && hasThreeOfAKind) {
+      return true
+    } else {
+      return false
+    }
+  }
+  
+  const hand1 = [5, 5, 5, 3, 3]
+  const hand2 = [5, 5, 3, 3, 4]
+  const hand3 = [5, 5, 5, 5, 4]
+  const hand4 = [7, 2, 7, 2, 7]
+
+  console.log(pokerHand(hand1))
+  console.log(pokerHand(hand2))
+  console.log(pokerHand(hand3))
+  console.log(pokerHand(hand4))
+ 
+  
+  
+  
+  
+  
```

Comment:
Same comment as above.  I think you probably used Chatgpt or something.  Also, if you're using TDD, then you don't need to console.log your function calls.  The test should tell you that it works.  Just remember this going forward.  :)